knimin/lib/data_access.py

```diff
@@ -864,6 +864,17 @@ def format_survey_data(self, md, external_surveys=None, full=False):  # noqa
                     bc_info['ag_login_id'] + participant_name).hexdigest()
                 md[1][barcode]['PUBLIC'] = 'Yes'
 
+                # Convert finer grained IBD to coarser grained
+                ibd = md[1][barcode].get('IBD_DIAGNOSIS_REFINED',
+                                         'Unspecified')
+                if ibd != 'Unspecified':
+                    if ibd in {"Ileal Crohn's Disease",
+                               "Colonic Crohn's Disease",
+                               "Ileal and Colonic Crohn's Disease"}:
+                        md[1][barcode]['IBD_DIAGNOSIS'] = "Crohn's disease"
```

Comment:
I understand that you want to make sure that both columns are in sync. But before changing the results shouldn't you check that the previous value is 'Unspecified'? If the participant already provided an answer to the previous question we should maintain whatever they said - correct?